projects/plugins/jetpack/class.jetpack-gutenberg.php

```diff
@@ -1114,3 +1127,14 @@ public static function get_render_callback_with_availability_check( $slug, $rend
 		};
 	}
 }
+
+/*
+ * Enable upgrade nudge for Atomic sites.
+ * This feature is false as default,
+ * so let's enable it through this filter.
+ *
+ * More doc: https://github.com/Automattic/jetpack/tree/master/projects/plugins/jetpack/extensions#upgrades-for-blocks
+ */
+if ( defined( 'IS_ATOMIC' ) && IS_ATOMIC ) {
```

Comment:
It could work too, I think. I opted by checking the global constant because it doesn't depend on the Site options. It should be faster. We already do something similar to check if it's a simple site `defined( 'IS_WPCOM' ) && IS_WPCOM;`.

Reply:
If it is faster, would it be worth updating `jetpack_is_atomic_site()` to rely on that too?

Reply:
I'm not sure the context where `jetpack_is_atomic_site()` is used in detail. I think it could be, though. Going to take a look at the code. IS_ATOMIC is mostly about the platform, while `jetpack_is_atomic_site()` seems to be tied to the site data.

Reply:
Is there a chance that `jetpack_is_atomic_site()` was introduced before `defined( 'IS_ATOMIC' )`? I can see only one _potential future_ downside in updating `jetpack_is_atomic_site()` to rely on that:
- A site goes to to Atomic -> true
- A site reverts to WPCOM -> false
but that same site was once an "automate transfer". Haven't found evidence that we remove `at_options` if site is transferred back to WPCOM but I think that currently transferring back to WPCOM involves a DB cleanup. 

So `jetpack_is_atomic_site()`_could_ be used in the future for that purposes.

That said, I still believe we should update `jetpack_is_atomic_site` to use `defined( 'IS_ATOMIC' ) && IS_ATOMIC` 😜

Reply:
Yes, IS_ATOMIC doesn't appear to be used currently in the Jetpack codebase. It would seem best to use `jetpack_is_atomic_site` for consistency, and then we can update the definition there if we agree that it should be checking the IS_ATOMIC constant and not the site options.

Reply:
addressed here 67f0b5a8a06b38af24df36f9b62a9a24ffe05df9